nisystemlink/clients/spec/utilities/_dataframe_utility.py

```diff
@@ -0,0 +1,266 @@
+    """Generate column header for a condition.
+
+    Args:
+        condition: Condition object for generating column header.
+
+    Returns:
+        The column header for the given condition.
+    """
+    name = condition.name or ""
+    unit = (
+        f"({condition.value.unit})"
+        if isinstance(condition.value, NumericConditionValue) and condition.value.unit
+        else ""
+    )
+
+    return f"condition_{name}{unit}"
+
+
+def __serialize_numeric_condition_range(value: NumericConditionValue) -> List[str]:
+    """Serialize ranges of a numeric condition value.
+
+    Args:
+        value: A condition's value with NumericConditionValue type.
+
+    Returns:
+        The list of ranges of the given value in a specific format.
+    """
+    if not value.range:
+        return []
+
+    return [
+        f"[{'; '.join(f'{k}: {v}' for k, v in vars(range).items() if v is not None)}]"
+        for range in value.range
+    ]
+
+
+def __serialize_condition_discrete_values(
+    value: Union[NumericConditionValue, StringConditionValue]
+) -> List[str]:
+    """Serialize discrete values of a value.
+
+    Args:
+        value: A condition's value with either NumericConditionValue type or StringConditionValue type.
+
+    Returns:
+        The list of discrete values of the given value in a string format.
+    """
+    return [str(discrete) for discrete in (value.discrete or [])]
+
+
+def __get_condition_values(condition: Condition) -> List[str]:
+    """Get ranges and discrete values of a condition.
+
+    Args:
+        condition: Condition for getting values.
+
+    Returns:
+        The list of values of the given condition in a specific format.
+    """
+    if not condition.value:
+        return []
+
+    values = []
+
+    if isinstance(condition.value, NumericConditionValue):
+        values.extend(__serialize_numeric_condition_range(value=condition.value))
+
+    values.extend(__serialize_condition_discrete_values(value=condition.value))
+
+    return values
+
+
+def __serialize_conditions(conditions: List[Condition]) -> Dict[str, str]:
+    """Serialize conditions into desired format.
+
+    Args:
+        conditions: List of all conditions in a spec.
+
+    Returns:
+        Conditions as a dictionary.
+    """
+    return {
+        __generate_condition_column_header(condition): ", ".join(
+            __get_condition_values(condition)
+        )
+        for condition in conditions
+    }
+
+
+def __format_specs_columns(specs_df: pd.DataFrame) -> pd.DataFrame:
+    """Format specs column to group conditions and keep properties adn keywords at the end.
+
+    Args:
+        specs_df: Dataframe of specs.
+
+    Returns:
+        Formatted dataframe of specs.
+    """
+    column_headers = specs_df.columns.to_list()
+    formatted_column_headers = [
+        header
+        for header in column_headers
+        if "condition_" not in header
+        and "properties." not in header
+        and "keywords" not in header
+    ]
```

Comment:
```suggestion
    formatted_column_headers = [
        header for header in column_headers
        if not any(substring in header for substring in ["condition_", "properties.", "keywords"])
    ]
```
can you check if this is working

Reply:
It's working.